src/veil/ringct/anonwallet.cpp

```diff
@@ -4025,13 +4025,18 @@ bool AnonWallet::UnlockWallet(const CExtKey& keyMasterIn)
     if (!SetMasterKey(keyMasterIn))
         return false;
 
+    {
     // Create stealth change account if it does not exist
     AnonWalletDB wdb(*walletDatabase);
     if (!wdb.ReadNamedExtKeyId("stealthchange", idChangeAccount)) {
         //If the change account is not created yet, then create it
         if (!CreateStealthChangeAccount(&wdb))
             return error("%s: failed to create stealth change account ", __func__);
     }
+    }
+
+    LOCK2(cs_main, pwalletParent->cs_wallet);
+    RescanWallet();
```

Comment:
This may be a good opportunity to mark RescanWallet with `EXCLUSIVE_LOCKS_REQUIRED` annotations if it always needs these locks, that way clang can catch issues like that in the future.